tests/research/test_batch02_durable_evidence_retention.py

```
@@ -0,0 +1,1239 @@
+    with pytest.raises(Exception):
+        persist_batch02_retained_result(
+            result_path,
+            {"status": "retry"},
+            run_context=ctx,
+        )
+    assert result_path.read_bytes() == original
+
+
+def test_historical_persist_rejects_b2_03_and_retained_persist_rejects_b2_02(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
```

Ruff (0.16.3)

[warning] Unused function argument: `monkeypatch`

(ARG001)